syntax = "proto3";
package population.proto;

import "games/market/proto/goods.proto";

message ConsumptionPackage {
  // Resources used up.
  market.proto.Container consumed = 1;

  // Resources required, but not consumed.
  market.proto.Container capital = 2;

  // You must be at least this [tag] to use the package.
  market.proto.Container required_tags = 4;

  // Tags acquired by consuming this package.
  market.proto.Container tags = 5;
}

message ConsumptionLevel {
  // Different options for this consumption level.
  repeated ConsumptionPackage packages = 1;
  // Tags acquired by consuming at this level.
  market.proto.Container tags = 2;
  // Name for easy reference.
  string name = 3;
}

message AutoProduction {
  market.proto.Container output = 1;
  market.proto.Container required_tags = 2;
}

message PopUnit {
  uint64 pop_id = 1;
  repeated int32 males = 2;
  repeated int32 women = 3;
  market.proto.Container wealth = 4;

  // Tags include education level and culture.
  market.proto.Container tags = 5;
}
